### PR TITLE
Fix preview links for DataObjects when using generators

### PR DESCRIPTION
**Pull request by jremmurd**
Using a LinkGenerator or PreviewGenerator service, the link shoudl be seen as "final" and no additional replacements should take place in the url string. Otherwise the link will most likely be invalid because auf double encoding the % sign.

Steps to reproduce:
1. add a link generator to any object
2. generate a link with escaped query parameters i.e. `/category/blaue%20schuhe~123456`
3. the preview will not work because the link was double quoted i.e. `/category/blaue%2520schuhe~123456`

### CODE REVIEW

**Comment by lukmzig** [comment]
Hi @jremmurd, first of all thank you for the contribution. 
I see you point why this might be problematic for your case but there also currently might be some custom Link generator / Preview generator implementations which rely on this behavior. If we would now implement this change it could be breaking for these.

**Comment by jremmurd** [comment]
> Hi @jremmurd, first of all thank you for the contribution. I see you point why this might be problematic for your case but there also currently might be some custom Link generator / Preview generator implementations which rely on this behavior. If we would now implement this change it could be breaking for these.

Understand, to me this behavior was a bug, so I provided a PR. Maybe just move it to another branch / version so we can ignore the BC break in the logic? Alternatively you could introduce an event to manipulate the preview url and skip the replacement of the `%` ?

I already applied this patch for our project, because the preview won't work otherwise.

**Comment by brusch** [comment]
Just another idea that could be (less) breaking: 
`$url = preg_replace('/%(?![0-9A-F]{2})/i', '%25', $url);`

WDYT? 
TL;DR: we just replace `%` which are not an URL encoded character ...

**Comment by jremmurd** [comment]
@brusch sounds good to me

File: bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php [hunk 1]
```diff
@@ -2046,9 +2046,9 @@ public function previewAction(Request $request)
                 return new Response("Preview not available, it seems that there's a problem with this object.");
             }
 
-            // replace all remainaing % signs
-            $url = str_replace('%', '%25', $url);
-
+            // replace all remainaing % signs which are not an url encoded character
```
Comment (brusch) [inline bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php]:
```suggestion
            // replace all remaining % signs which are not an url encoded characters
```

**Comment by lukmzig** [comment]
@jremmurd could you please kindly rebase your changes to 10.6 branch as it can still potentially be a breaking change? Thank you very much!

**Comment by lukmzig** [comment]
thanks @jremmurd
